src/psyclone/tests/generator_test.py

```diff
@@ -417,41 +460,43 @@ def test_profile_gocean():
     Profiler._options = []
 
 
-def test_script_attr_error():
+def test_script_attr_error(script_factory):
     '''Checks that generator.py raises an appropriate error when a script
-    file contains a trans() function which raises an attribute
-    error. This is what we previously used to check for a script file
-    not containing a trans() function.
+    file contains a trans() function which raises an attribute error.
 
     '''
+    error_script = script_factory("""
+from psyclone.psyGen import Loop
+from psyclone.transformations import ColourTrans
+
+def trans(psyir):
+    ''' A valid trans function which produces an attribute error as
+    we have mistyped apply()'''
+    ctrans = ColourTrans()
+    for child in psyir.walk(Loop):
+        if isinstance(child, Loop) and child.field_space != "w3":
+            ctrans.appy(child)
+""")
     with pytest.raises(Exception) as excinfo:
         _, _ = generate(os.path.join(BASE_PATH, "dynamo0p3",
                                      "1_single_invoke.f90"),
-                        api="lfric",
-                        script_name=os.path.join(BASE_PATH,
-                                                 "dynamo0p3",
-                                                 "error_trans.py"))
+                        api="lfric", script_name=error_script)
     assert 'object has no attribute' in str(excinfo.value)
 
 
-def test_script_null_trans():
+def test_script_null_trans(script_factory):
     '''Checks that generator.py works correctly when the trans() function
-    in a valid script file does no transformations (it simply passes
-    input to output). In this case the valid script file has an
-    explicit path and must therefore exist at this location.
+    in a valid script file does no transformations. In this case the valid
```

Comment:
Can probably remove the last sentence as there's no path manipulation here any more.